src/Workspaces/MSBuildTest/OpenProjectsTests.cs

```diff
@@ -0,0 +1,222 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.MSBuild.UnitTests
+{
+    // Generate a list of templates by running `dotnet new list --type project`
+    public class OpenProjectsTests : MSBuildWorkspaceTestBase
+    {
+        // Microsoft.CodeAnalysis.VisualBasic.ERRID
+        private const string ERR_UndefinedType1 = "BC30002";
+
+        protected ITestOutputHelper TestOutputHelper { get; set; }
+
+        public OpenProjectsTests(ITestOutputHelper output)
+        {
+            TestOutputHelper = output;
+        }
+
+        [ConditionalTheory(typeof(DotNetSdkMSBuildInstalled), typeof(WindowsOnly))]
+        [InlineData("winforms")]
+        [InlineData("winformslib")]
+        [InlineData("winformscontrollib")]
+        [InlineData("wpf")]
+        [InlineData("wpflib")]
+        [InlineData("wpfcustomcontrollib")]
+        [InlineData("wpfusercontrollib")]
+        public async Task CSharpTemplateProject_WindowsOnly_LoadWithNoDiagnostics(string templateName)
+        {
+            var ignoredDiagnostics = templateName switch
+            {
+                _ => Array.Empty<string>(),
+            };
+
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.CSharp, ignoredDiagnostics);
+        }
+
+        [ConditionalTheory(typeof(DotNetSdkMSBuildInstalled))]
+        [InlineData("api")]
+        [InlineData("web")]
+        [InlineData("grpc")]
+        [InlineData("webapi")]
+        [InlineData("webapp")]
+        [InlineData("mvc")]
+        [InlineData("angular")]
+        [InlineData("react")]
+        [InlineData("blazor")]
+        [InlineData("blazorwasm")]
+        [InlineData("classlib")]
+        [InlineData("console")]
+        [InlineData("mstest")]
+        [InlineData("nunit")]
+        [InlineData("razorclasslib")]
+        [InlineData("worker")]
+        [InlineData("xunit")]
```

Comment:
Consider sorting this.

Reply:
Sure, this is the order listed by dotnet CLI. If we continue to manually maintain the list, this sorting might make it easier.

Reply:
Moved to generating this list.